src/Storages/MergeTree/MergeTreeSequentialSource.cpp

```diff
@@ -56,6 +63,8 @@ class MergeTreeSequentialSource : public ISource
 
     Poco::Logger * log = &Poco::Logger::get("MergeTreeSequentialSource");
 
+    std::optional<MarkRanges> mark_ranges;
```

Comment:
Need a comment, in which case it will be initialized (mutation?) and when it is empty (merge?)